distributed/tests/test_steal.py

```diff
@@ -1841,27 +1848,20 @@ def _get_task_placement(
     actual = []
     for w in workers:
         actual.append(
-            [list(key_split(ts.key)) for ts in s.workers[w.address].processing]
+            [
+                list(key_split(key[5:]))  # Remove "task-" prefix
+                for key in w.data.keys()
+                if key.startswith("task-")
+            ]
```

Comment:
Could you explain the reasoning behind this change?

Reply:
The tests should now check where the tasks have eventually been executed after stealing took place. This is the behavior we want to test here and relies less on internals as the previous version. The latter relied on checking the scheduler's worker state after stealing but during processing.